### PR TITLE
fcrypt: consider the environment (TMPDIR) when looking for the tempor…

### PR DESCRIPTION
**Pull request by petermax2**
# Purpose

…ary directory

See #1973 for further details

@ingwinlu Does this fix your problem described in #1973 ?

# Checklist

Check relevant points but please do not remove entries.
For docu fixes, spell checking, and similar nothing
needs to be checked.

- [x] commit messages are fine ("module: short statement" syntax and refer to issues)
- [ ] I added unit tests
- [x] I ran all tests locally and everything went fine
- [x] affected documentation is fixed
- [x] I added code comments, logging, and assertions (see doc/CODING.md)
- [ ] meta data is updated (e.g. README.md of plugins)
- [x] release notes are updated (doc/news/_preparation_next_release.md)

@markus2330 please review my pull request

### CODE REVIEW

File: src/plugins/fcrypt/fcrypt.c [hunk 1]
```diff
@@ -64,13 +64,44 @@ typedef struct _fcryptState fcryptState;
...
 		tmpDir = keyString (k);
 	}
 
+	if (!tmpDir)
+	{
+		// check the environment; returns NULL if no match is found
+		tmpDir = getenv ("TMPDIR");
+	}
+
+	if (!tmpDir)
```
Comment (markus2330) [inline src/plugins/fcrypt/fcrypt.c]: Imho TMPDIR is enough. We should not use non-standard vars.

**Comment by petermax2** [comment]
@sanssecours could you please restart the failed Travis job again?

    Failed to download resource "qt"

Strange that it always fails at Qt. Maybe the package is too big and out-of-cache or something?

**Comment by petermax2** [comment]
jenkins build all please

**Comment by sanssecours** [comment]
> could you please restart the failed Travis job again?

I just did. However, you can do that yourself too. Just sign up at https://travis-ci.org with your GitHub account. Afterwards you should be able to restart all Travis jobs for this repository.

**Comment by petermax2** [comment]
@markus2330 ready to merge 🎉

**Comment by markus2330** [comment]
Thank you for the fix!

**Pull request merged**